DependencyInjection/Configuration.php

```diff
@@ -82,8 +82,21 @@ private function addDynamicSection(ArrayNodeDefinition $root)
                             ->prototype('scalar')->end()
                         ->end() // controllers_by_type
                         ->arrayNode('controllers_by_class')
-                            ->useAttributeAsKey('class')
-                            ->prototype('scalar')->end()
+                            ->prototype('array')
+                                ->ignoreExtraKeys(false)
```

Comment:
This one was the only chance to get a nested array without an own key, but now i got no validation on deeper key names (value, methods) anymore.